src/conversion.c

```diff
@@ -844,3 +844,93 @@ int igraph_get_stochastic_sparsemat(const igraph_t *graph,
   
   return 0;
 }
+
+
+/**
+ * \ingroup conversion
+ * \function igraph_to_prufer
+ * \brief Converts a tree to its Pr&uuml;fer sequence
+ *
+ * A Pr&uuml;fer sequence is a unique sequence of integers associated
+ * with a labelled tree. A tree on n >= 2 vertices can be represented by a
+ * sequence of n-2 integers, each between 0 and n-1 (inclusive).
+ *
+ * \param graph Pointer to an initialized graph object which
+          must be a tree on n >= 2 vertices.
+ * \param prufer A pointer to the integer vector that should hold the Pr&uuml;fer sequence;
+                 the vector must be initialized and will be resized to n - 2.
+ * \return Error code:
+ *          \clist
+ *          \cli IGRAPH_ENOMEM
+ *             there is not enough memory to perform the operation.
+ *          \cli IGRAPH_EINVAL
+ *             the graph has less than 2 vertices.
+ *          \endclist
+ *
+ * \sa \ref igraph_tree(), \ref igraph_from_prufer()
+ *
+ */
+int igraph_to_prufer(const igraph_t *graph, igraph_vector_int_t* prufer) {
+  /* For generating the Prüfer sequence, we enumerate the vertices u of the tree.
+     We keep track of the degrees of all vertices, treating vertices
+     of degree 0 as removed. We maintain the invariant that all leafs
+     that are still contained in the tree are >= u.
+     If u is a leaf, we remove it and add its unique neighbor to the prüfer
+     sequence. If the removal of u turns the neighbor into a leaf which is < u,
+     we repeat the procedure for the new leaf and so on. */
+  igraph_integer_t u;
+  igraph_vector_t degrees, neighbors;
+  igraph_integer_t prufer_index = 0;
+  igraph_integer_t n = igraph_vcount(graph);
+
+  if(n < 2)
+    return IGRAPH_EINVAL;
```

Comment:
On error, do not simply return, use `IGRAPH_ERROR` instead and report a user-readable error. `IGRAPH_ERROR` also ensures that the error handler will be called and any allocated resources can be freed (should a future version of this function allocate resources before this line).